sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuShuffledHashJoinExec.scala

```diff
@@ -290,7 +296,7 @@ object GpuShuffledHashJoinExec extends Logging {
       var isBuildSerialized = false
       // Batches type detection
       val coalesceBuiltIter = getHostShuffleCoalesceIterator(
-        bufBuildIter, targetSize, coalesceMetrics).map { iter =>
+        bufBuildIter, buildDataType, targetSize, coalesceMetrics).map { iter =>
```

Comment:
The build side schema is already defined by `buildTypes` at line #293. So `buildDataType` is not necessary.